### PR TITLE
209/order checkboxes

### PR DESCRIPTION
**Pull request by alfetopito**
Part of #209 
Depends on #327 

---

- Adding logic to checkboxes

## Next PR:
- Order cancellation

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr333--dexreact.review.gnosisdev.com

File: src/components/OrdersWidget/index.tsx [hunk 1]
```diff
@@ -206,7 +227,7 @@ const OrdersWidget: React.FC = () => {
             <div className="ordersContainer">
               <div className="headerRow">
                 <div className="checked">
-                  <input type="checkbox" />
+                  <input type="checkbox" onClick={toggleSelectAll} checked={orders.length === markedForDeletion.size} />
```
Comment (Velenir) [inline src/components/OrdersWidget/index.tsx]: It's more common to use `onChange` instead of `onClick` for checkboxes and radio elements. I can't say if one is better than the other, just more common

File: src/components/OrdersWidget/OrderRow.tsx [hunk 2]
```diff
@@ -73,6 +73,21 @@ const PendingLink: React.FC<Pick<Props, 'pending'>> = ({ pending }) => {
...
 
+const DeleteOrder: React.FC<Pick<Props, 'isMarkedForDeletion' | 'toggleMarkedForDeletion' | 'pending'>> = ({
+  isMarkedForDeletion,
+  toggleMarkedForDeletion,
+  pending,
+}) => (
+  <div className="checked">
+    <input
+      type="checkbox"
+      onClick={toggleMarkedForDeletion}
```
Comment (Velenir) [inline src/components/OrdersWidget/OrderRow.tsx]: See below about `onChange`

File: src/api/index.ts [hunk 3]
```diff
@@ -16,16 +16,34 @@ import {
...
 import { INITIAL_INFURA_ENDPOINT } from 'const'
 
+const isWeb3Mock = process.env.MOCK_WEB3 === 'true'
 const isWalletMock = process.env.MOCK_WALLET === 'true'
 const isTokenListMock = process.env.MOCK_TOKEN_LIST === 'true'
 const isErc20Mock = process.env.MOCK_ERC20 === 'true'
 const isDepositMock = process.env.MOCK_DEPOSIT === 'true'
 const isExchangeMock = process.env.MOCK_EXCHANGE === 'true'
 
+// TODO connect to mainnet if we need AUTOCONNECT at all
```
Comment (anxolin) [inline src/api/index.ts]: Why this change is in this PR?

**Comment by anxolin** [comment]
Well, there's an issue with the layout. The checkboxes are moving from their position when you check-uncheck them, and that's a bit inconvenient:

image

Actually, this is why in my original proposal I was suggesting to always show the message, cause it was good enough imo and we prevent this cases. Anyways, now is done I would try to solve the layout.

One easy solution is using CSS (visibility), instead of display: none, or not rendering the message

File: src/api/index.ts (hunk 3 above)
Comment (alfetopito) [reply src/api/index.ts]: It's not. It was tracking another branch.
